plantcv/plantcv/transform/warp.py

```diff
@@ -1,58 +1,82 @@
 # warp image
 
 import cv2
+import skimage
 import os
 import numpy as np
-from plantcv.plantcv import print_image
-from plantcv.plantcv import plot_image
+import copy
+from plantcv.plantcv import params
+from plantcv.plantcv._debug import _debug
 from plantcv.plantcv import fatal_error
 from plantcv.plantcv import color_palette
-from plantcv.plantcv import params
 from plantcv.plantcv.visualize import overlay_two_imgs
+from plantcv.plantcv.transform import rescale
+from skimage import img_as_ubyte
 
+def _preprocess_img_dtype(img):
```

Comment:
do we need this function?? I think you can just send img through `img_as_ubyte()` and it will return what we need.  also rescale doesn't work on rgb images which are supported in  the rest of the function. one caveat is that the scikit-image function will only accept float between -1,1 and will clip negative numbers upon conversion to uint8. I think we can live with that though. most float data would be between 0-1 for images(?) and mostly all plantcv functions only work with uint8

Reply:
or proceed img_as_ubyte() with 
if img.dtype == 'uint8':

Reply:
The values for thermal images won't be between -1,1, so `pcv.transform.rescale` is used to convert the datatype. To make this function more general, use `skimage.img_as_ubyte` to catch `RuntimeError`. The assumption here is that if an image is a multi-channel one, values are between -1,1.